proto/buf/alpha/registry/v1alpha1/schema.proto

```diff
@@ -0,0 +1,196 @@
+// Copyright 2020-2022 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package buf.alpha.registry.v1alpha1;
+
+import "google/protobuf/descriptor.proto";
+
+// Provides endpoints for downloading dynamic schemas and for using schemas
+// in validation and data transformation functions.
+service SchemaService {
+  // GetSchema allows the caller to download a schema for one or more requested
+  // types, RPC services, or RPC methods.
+  rpc GetSchema(GetSchemaRequest) returns (GetSchemaResponse);
+
+  // Reformat allows the caller to indicate a type schema which will be used to
+  // transform given data to a different format.
+  rpc Reformat(ReformatRequest) returns (ReformatResponse);
+}
+
+// Module identifies a particular module in the registry.
+message Module {
+  // A user or organization. Required.
+  string owner = 1;
+  // A repository or module name. Required.
+  string repository = 2;
+
+  // A reference to a version. Optional. If present, it may refer to either a
+  // tag or a commit.
+  //
+  // If absent, the latest version for the module identified by the above
+  // owner and repository fields is assumed.
+  string reference = 3;
```

Comment:
@smallsamantha Are the IDs for drafts also called "commits" or did we use a different term?

Reply:
If there is better vernacular to use for these fields, I'd love to hear it. In particular, I'd like to make sure we are using the same terms here as BSR public docs might be using, to eliminate confusion in the API over terminology.

But I used these field names because they are consistent with how we refer to a version of a module today in [these](https://github.com/bufbuild/buf/blob/4b1c53e9b38ee78dbacbf3b1dd3810eab7f0de95/proto/buf/alpha/registry/v1alpha1/image.proto#L47-L49) [three](https://github.com/bufbuild/buf/blob/4b1c53e9b38ee78dbacbf3b1dd3810eab7f0de95/proto/buf/alpha/registry/v1alpha1/module.proto#L25-L28) [places](https://github.com/bufbuild/buf/blob/4b1c53e9b38ee78dbacbf3b1dd3810eab7f0de95/proto/buf/alpha/module/v1alpha1/module.proto#L57-L60).